### PR TITLE
[browser] Allow downloading WebAssembly resources without performing other WebAssembly runtime initalization

### LINKED ISSUE #100299: [browser] Allow downloading WebAssembly resources without performing other WebAssembly runtime initalization
Blazor has the ability to enable interactivity using SignalR while initializing the WebAssembly runtime in the background. After WebAssembly assets get downloaded, future visits to the site will use WebAssembly for interactivity, since they will load quickly from the cache.

However, in addition to downloading WebAssembly resources, the `dotnet.create()` JS API also performs other initialization that blocks the main thread (for about ~85 ms on my machine). This can contribute to the webpage feeling sluggish.

If we had an API like `dotnet.download()` that _only_ downloaded WebAssembly resources and skipped other initialization, that could help the case where Blazor's "Auto" render mode uses Server interactivity.

### PR DESCRIPTION
**Pull request by pavelsavara**
Fixes https://github.com/dotnet/runtime/issues/100299

```
import { dotnet } from './_framework/dotnet.js'

// optional call to download all assets
await dotnet.download();

// and later it could be followed by usual
const runtime = await dotnet.create();
```

Note that when we download assets during cold start, they are being loaded into WASM memory as they arrive. 
When you already have pre-downloaded the assets with this new API, the startup of the runtime will be more of the burst. Similar as with assets already in a cache.

### CODE REVIEW

**Comment by maraf** [comment]
Please add a WBT. Since we will remove custom cache shortly, we can check that all URLs that are used during "download" phase will be used during real start phase.

File: src/mono/browser/runtime/loader/run.ts [hunk 1]
```diff
@@ -375,16 +384,22 @@ export class HostBuilder implements DotnetHostBuilder {
...
     }
     mono_assert(emscriptenModule, "Null moduleConfig");
     mono_assert(loaderHelpers.config, "Null moduleConfig.config");
-    await createEmscripten(emscriptenModule);
-    return globalObjectsRoot.api;
-}
 
-export async function createEmscripten (moduleFactory: DotnetModuleConfig | ((api: RuntimeAPI) => DotnetModuleConfig)): Promise<RuntimeAPI | EmscriptenModuleInternal> {
     // extract ModuleConfig
     if (typeof moduleFactory === "function") {
```
Comment (maraf) [inline src/mono/browser/runtime/loader/run.ts:404]: Why do we need to run moduleFactory if we want to only download resources?

Comment (pavelsavara) [reply src/mono/browser/runtime/loader/run.ts:404]: It could change the location of the config file

**Comment by pavelsavara** [comment]
> Please add a WBT. Since we will remove custom cache shortly, we can check that all URLs that are used during "download" phase will be used during real start phase.

I'm not sure that I follow. What would the test look like ?

**Comment by maraf** [comment]
> I'm not sure that I follow. What would the test look like ?

What I'm thinking about for the test is that we can ensure that no new resource will be requested during real start after `download()` was called. So the idea is to install `resourceLoader` (or override `fetch`), record all URLs used during `download()` and verify that no new URLs were requested during `create()`

**Review by maraf — APPROVED** [review]

**Pull request merged**